src/streaming/array.cpp

```diff
@@ -140,60 +67,74 @@ zarr::ArrayWriter::write_frame(std::span<const std::byte> data)
     return bytes_written;
 }
 
+bool
+zarr::Array::close_()
+{
+    bool retval = false;
+    is_closing_ = true;
```

Comment:
`is_closing_` looks like it's only read in `v3.array.cpp`. Is that right? 

I'm sort of worried about a race condition on is_closing_ in compress_and_flush